src/renderers/common/extras/PMREMGenerator.js

```diff
@@ -593,39 +603,65 @@ class PMREMGenerator {
 			pingPongRenderTarget,
 			lodIn,
 			lodOut,
-			sigma,
-			'latitudinal',
-			poleAxis );
+			'latitudinal' );
 
 		this._halfBlur(
 			pingPongRenderTarget,
 			cubeUVRenderTarget,
 			lodOut,
 			lodOut,
-			sigma,
-			'longitudinal',
-			poleAxis );
+			'longitudinal' );
 
 	}
 
-	_halfBlur( targetIn, targetOut, lodIn, lodOut, sigmaRadians, direction, poleAxis ) {
+	_halfBlur( targetIn, targetOut, lodIn, lodOut, direction ) {
 
 		const renderer = this._renderer;
 		const blurMaterial = this._blurMaterial;
 
-		if ( direction !== 'latitudinal' && direction !== 'longitudinal' ) {
+		const blurMesh = this._lodMeshes[ lodOut ];
+
+		blurMesh.material = blurMaterial;
+		blurMesh.userData = this._getUserDataBlur( targetIn, lodIn, lodOut, direction );
+
+		targetIn.texture.frame = ( targetIn.texture.frame || 0 ) + 1;
+
+		blurMaterial._envMap.value = targetIn.texture;
+
+		const { _lodMax } = this;
+
+		const outputSize = this._sizeLods[ lodOut ];
+		const x = 3 * outputSize * ( lodOut > _lodMax - LOD_MIN ? lodOut - _lodMax + LOD_MIN : 0 );
+		const y = 4 * ( this._cubeSize - outputSize );
+
+		_setViewport( targetOut, x, y, 3 * outputSize, 2 * outputSize );
+		renderer.setRenderTarget( targetOut );
+		renderer.render( blurMesh, _flatCamera );
+
+	}
+
+	_getUserDataBlur( targetIn, lodIn, lodOut, direction ) {
+
+		const cache = this._userData[ direction ];
+
+		if ( cache === undefined ) {
 
 			console.error( 'blur direction must be either latitudinal or longitudinal!' );
 
 		}
 
-		// Number of standard deviations at which to cut off the discrete approximation.
-		const STANDARD_DEVIATIONS = 3;
+		const u = cache[ lodIn ];
 
-		const blurMesh = this._lodMeshes[ lodOut ];
-		blurMesh.material = blurMaterial;
+		if ( u !== undefined ) return u;
+
+		// populate data for this pass
 
-		const blurUniforms = blurMaterial.uniforms;
+		const { _lodMax } = this;
+
+		const sigmaRadians = Math.sqrt( this._sigmas[ lodOut ] * this._sigmas[ lodOut ] - this._sigmas[ lodOut - 1 ] * this._sigmas[ lodOut - 1 ] );
```

Comment:
Hmm - is this right? The cache is based on `lodIn`, but the sigmas we're saving are based on `lodOut`. Sometimes they are the same and sometimes they are not, based on how halfblur is called. And does this all continue to work when a PMREM is generated with a different size?